src/stan/error_handling/out_of_range.hpp

```diff
@@ -0,0 +1,97 @@
+#ifndef STAN__ERROR_HANDLING__OUT_OF_RANGE_HPP
+#define STAN__ERROR_HANDLING__OUT_OF_RANGE_HPP
+
+#include <typeinfo>
+#include <string>
+#include <sstream>
+#include <stdexcept>
+#include <stan/meta/traits.hpp>
+
+namespace stan {
+  namespace error_handling {
+
+    /**
+     * Throw an out_of_range exception with a consistently formatted message.
+     * 
+     * This is an abstraction for all Stan functions to use when throwing
+     * out of range. This will allow us to change the behavior for all
+     * functions at once.
+     *
+     * The message is:
```

Comment:
I left it doc'ed individually because are only a handful of these functions and when I was looking at it, it was easier to use with the doc next to the function. If we end up with a lot more, we can point.